internal/runbits/progressbar.go

```diff
@@ -0,0 +1,222 @@
+package runbits
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/pkg/platform/runtime/artifact"
+	"github.com/ActiveState/cli/pkg/platform/runtime/setup/events"
+	"github.com/vbauerster/mpb/v6"
+	"github.com/vbauerster/mpb/v6/decor"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+type artifactStepBar struct {
+	started time.Time
+	bar     *mpb.Bar
+}
+
+// progressBar receives update events and modifies a global state accordingly
+type progressBar struct {
```

Comment:
I feel like this name is a bit missleading. Maybe instead call it `runtimeProgress` or something?

Reply:
Additionally looking at the methods on `progressBar` this feels really messy atm. It's not clear what `progressBar`'s responsibility is.

At the moment it does at least 3 different thing:

 - Handle runtime messages
 - Track progressbar state
 - Provide helper utilities (like trimName)

It would be nice if you could clean this up significantly. I'd like to be able to look at just the method names and get a good understanding of how it operates.

Reply:
As explained in my comment on the `events` package, this file (as it is in the runbits package) is focused on the output part, so to actually print the progress bars.  Therefore, it needs to keep the `mpb.Bar` pointers, which hold output-related state.
It is a little unfortunate, that I had to add the `artifactStepState` (I should rename it into `artifactStepBar`), because the `mpb.Bar` does not capture the time when it started, but wants me to tell it with every update.  I believe that this optimally should be stored in the `mpb.Bar` itself, so it is done here.  

Other Action items based on your comments:
- I will split out the `trimWidth` and `addTotalBar`, `addProgressBar` implementations into separate file, as they are more specific, but I think that they should still be part of the same struct.  As the entire file describes how to display the runtime progress.
- I will remove the dependency on `events.ArtifactsSetupStep` by moving the translation from the `Step` to a "title" into the `events` package.  That is certainly cleaner.